src/SessionWidget.vala

```diff
@@ -107,6 +140,82 @@ public class A11Y.SessionWidget : Gtk.Grid {
         mouse_settings.bind ("dwell-click-enabled", hover_click, "active", SettingsBindFlags.DEFAULT);
     }
 
+    ~SessionWidget () {
+        if (keyboard_pid != 0) {
+            Posix.kill (keyboard_pid, Posix.Signal.KILL);
+            Posix.waitpid (keyboard_pid, out status, 0);
+            keyboard_pid = 0;
+        }
+
+        if (reader_pid != 0) {
+            Posix.kill (reader_pid, Posix.Signal.KILL);
+            Posix.waitpid (reader_pid, out status, 0);
+            reader_pid = 0;
+        }
+    }
+
+    private void toggle_screen_reader (bool active) {
```

Comment:
I would name (or comment) this to indicate it is only called in a greeter session.  May add `requires settings != null`.